x-pack/plugins/security_solution/public/detection_engine/rule_creation_ui/pages/rule_creation/index.tsx

```diff
@@ -285,16 +289,28 @@ const CreateRulePageComponent: React.FC = () => {
             stepIsValid(scheduleStep) &&
             stepIsValid(actionsStep)
           ) {
-            const createdRule = await createRule(
-              formatRule<RuleCreateProps>(
-                defineStep.data,
-                aboutStep.data,
-                scheduleStep.data,
-                actionsStep.data
-              )
-            );
-
-            displaySuccessToast(i18n.SUCCESSFULLY_CREATED_RULES(createdRule.name), dispatchToaster);
+            const startMlJobsIfNeeded = async () => {
+              if (!isMlRule(defineStep.data.ruleType) || !actionsStep.data.enabled) {
+                return;
+              }
+              setIsStartingJobs(true);
+              await startMlJobs(defineStep.data.machineLearningJobId);
+              setIsStartingJobs(false);
+            };
+            const [, createdRule] = await Promise.all([
+              startMlJobsIfNeeded(),
+              createRule(
+                formatRule<RuleCreateProps>(
+                  defineStep.data,
+                  aboutStep.data,
+                  scheduleStep.data,
+                  actionsStep.data
+                )
+              ),
+            ]);
+
+            addSuccess(i18n.SUCCESSFULLY_CREATED_RULES(createdRule.name));
```

Comment:
Please mind that `createRule` doesn't throw - it shows an error toast in case of an underlying exception. This means that something is wrong with the error handling here: we call `addSuccess` and navigate to the Rule Management page regardless of the result of the rule creation API call.